spring/boot-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaClientHttpResponse.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.client.reactive.ClientHttpResponse;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.spring.web.reactive.ArmeriaHttpClientResponseSubscriber.ResponseBodyPublisher;
+
+import io.netty.handler.codec.http.cookie.ClientCookieDecoder;
+import reactor.core.publisher.Flux;
+
+/**
+ * A {@link ClientHttpResponse} implementation for the Armeria HTTP client.
+ */
+final class ArmeriaClientHttpResponse implements ClientHttpResponse {
+
+    private final com.linecorp.armeria.common.HttpStatus status;
+    private final com.linecorp.armeria.common.HttpHeaders headers;
+
+    private final Flux<DataBuffer> body;
+
+    @Nullable
+    private MultiValueMap<String, ResponseCookie> cookies;
+    @Nullable
+    private HttpHeaders httpHeaders;
+
+    ArmeriaClientHttpResponse(com.linecorp.armeria.common.HttpHeaders headers,
+                              ResponseBodyPublisher publisher,
+                              DataBufferFactoryWrapper<?> factoryWrapper) {
+        this.headers = requireNonNull(headers, "headers");
+        final com.linecorp.armeria.common.HttpStatus status = headers.status();
+        checkArgument(status != null, "no status in HTTP headers");
+        this.status = status;
+
+        body = Flux.from(publisher).cast(HttpData.class).map(factoryWrapper::toDataBuffer);
+    }
+
+    @Override
+    public HttpStatus getStatusCode() {
+        // Raise an IllegalArgumentException if the status code is unknown.
+        return HttpStatus.valueOf(status.code());
+    }
+
+    @Override
+    public int getRawStatusCode() {
+        // Return the status code even if it is unknown.
+        return status.code();
+    }
+
+    @Override
+    public MultiValueMap<String, ResponseCookie> getCookies() {
+        final MultiValueMap<String, ResponseCookie> cookies = this.cookies;
+        if (cookies != null) {
+            return cookies;
+        }
+        this.cookies = initCookies();
+        return this.cookies;
+    }
+
+    @Override
+    public Flux<DataBuffer> getBody() {
+        return body;
+    }
+
+    @Override
+    public HttpHeaders getHeaders() {
+        final HttpHeaders httpHeaders = this.httpHeaders;
+        if (httpHeaders != null) {
+            return httpHeaders;
+        }
+        this.httpHeaders = initHttpHeaders();
+        return this.httpHeaders;
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this).omitNullValues()
+                          .add("status", status)
+                          .add("headers", headers)
+                          .add("cookies", cookies)
+                          .add("httpHeaders", httpHeaders)
```

Comment:
Is this for a debugging purpose? I guess a user will see both `headers` and `httpHeaders` which may be confusing.